src/uri/encoding.cr

```diff
@@ -1,4 +1,58 @@
 class URI
+  # Encodes *string* so it can be safely placed as a potentially multi-segmented
+  # URI path, replacing special characters with URI escape sequences as needed.
+  #
+  # Unreserved characters such as ASCII letters, digits, and the characters
+  # `_.-~` are not encoded, as well as the character `/` which represent a
+  # segment separator in hierarchical paths ([RFC 3986 §3.3](https://datatracker.ietf.org/doc/html/rfc3986#section-3.3)).
+  #
+  # ```
+  # require "uri"
+  #
+  # URI.encode_path("foo/bar/baz")  # => "foo/bar/baz"
+  # URI.encode_path("hello world!") # => "hello%20world%21"
+  # URI.encode_path("put: it+й")    # => "put%3A%20it%2B%D0%B9"
+  # ```
+  #
+  # * `.decode` is the reverse operation.
+  # * `.encode_path_segment` encodes a single path segment, escaping `/`.
+  def self.encode_path(string : String) : String
+    String.build { |io| encode_path(io, string) }
+  end
+
+  # :ditto:
+  def self.encode_path(io : IO, string : String) : Nil
+    self.encode(string, io) { |byte| URI.unreserved?(byte) || '/' === byte }
+  end
+
+  # Encodes *string* so it can be safely placed inside a URI path segment,
+  # replacing special characters (including `/`) with URI escape sequences as needed.
+  #
+  # Unreserved characters such as ASCII letters, digits, and the characters
+  # `_.-~` are not encoded ([RFC 3986 §3.3](https://datatracker.ietf.org/doc/html/rfc3986#section-3.3)).
+  #
+  # ```
+  # require "uri"
+  #
+  # URI.encode_path_segment("foo;bar;baz")  # => "foo%3Bbar%3Bbaz"
+  # URI.encode_path_segment("foo/bar/baz")  # => "foo%2Fbar%2Fbaz"
+  # URI.encode_path_segment("foo,bar,baz")  # => "foo%2Cbar%2Cbaz"
+  # URI.encode_path_segment("hello world!") # => "hello%20world%21"
+  # URI.encode_path_segment("put: it+й")    # => "put%3A%20it%2B%D0%B9"
+  # ```
+  #
+  # * `.decode` is the reverse operation.
+  # * `.encode_path` encodes a path consisting of multiple segments, not escaping `/`.
+  # * `.encode_www_form` escapes space character as `+`.
```

Comment:
```suggestion
  # * To avoid escaping the `/` character, use `.encode_path`.
  # * To escape the space character as `+` instead, use `.encode_www_form`.
  #
  # To further customize the choice of escaped characters, fall back to the block-based variant
  # `.encode(string : String, io : IO, space_to_plus : Bool = false, &block)`.
  # For example, to also escape the comma but avoid escaping the exclamation mark:
  #
  # ```
  # string = "hello, world!"
  # String.build do |io|
  #   URI.encode(string, io) do |byte|
  #     (URI.unreserved?(byte) || byte == '!'.ord) && byte != ','.ord
  #   end
  # end # => "hello%2C%20world!"
  # ```
```

Reply:
I don't think this example fits very well in the documentation of `.encode_path_segment`. It illustrates how to use a different method for a different use case. It would better fit in the documentation of `.encode` or the *Encoding* section in the class doc.